controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -407,4 +412,51 @@ public void updateStreamState(final String scopeName, final String streamName,
         }).thenApply(asyncResponse::resume)
                 .thenAccept(x -> LoggerHelpers.traceLeave(log, "updateStreamState", traceId));
     }
+
+    /**
+     * Implementation of getScalingEvents REST API.
+     *
+     * @param scopeName         The scope name of stream.
+     * @param streamName        The name of stream.
+     * @param from              DateTime from which scaling events should be displayed.
+     * @param to                DateTime until which scaling events should be displayed.
+     * @param securityContext   The security for API access.
+     * @param asyncResponse     AsyncResponse provides means for asynchronous server side response processing.
+     */
+    @Override
+    public void getScalingEvents(final String scopeName, final String streamName, final Long from, final Long to,
+                                 final SecurityContext securityContext, final AsyncResponse asyncResponse) {
+        long traceId = LoggerHelpers.traceEnter(log, "getScalingEvents");
+
+        controllerService.getScaleRecords(scopeName, streamName).thenApply(listScaleMetadata -> {
+            Iterator<ScaleMetadata> metadataIterator = listScaleMetadata.iterator();
+            List<ScaleMetadata> finalScaleMetadataList = new ArrayList<ScaleMetadata>();
+            ScaleMetadata referenceEvent = null;
+            while (metadataIterator.hasNext()) {
+                ScaleMetadata scaleMetadata = metadataIterator.next();
+                if (scaleMetadata.getTimestamp() >= from && scaleMetadata.getTimestamp() <= to) {
+                    finalScaleMetadataList.add(scaleMetadata);
+                } else if (scaleMetadata.getTimestamp() < from && scaleMetadata.getTimestamp() < to) {
```

Comment:
From the requirement here - https://github.com/pravega/pravega/issues/1347, seems like we need the initial state of the stream -
"Should return a “scaling event” that describes the initial state of the stream when created. Will be the only entry returned for static segment streams"
But here we might be returning any arbitrary event. Also please review if this event is guaranateed to be present, cc @shiveshr.

Remove the redundant "< to" check